testsuite/tests/apicast/parameters/test_modular_apicast.py

```diff
@@ -45,34 +36,44 @@ def build_images(openshift, request, image_template,
     """
     openshift_client = openshift()
 
+    github_template = resources.files('testsuite.resources.modular_apicast').joinpath("example_policy.yml")
+    copy_template = resources.files('testsuite.resources.modular_apicast').joinpath("example_policy_copy.yml")
+
     amp_release = openshift_client.image_stream_tag("amp-apicast")
-    build_name_example_policy = blame(request, "apicast-example-policy")
-    build_name_custom_policies = blame(request, "apicast-custom-policies")
-    image_stream_apicast_policy = blame(request, "is-apicast-policy")
+    build_name_github = blame(request, "apicast-example-policy-github")
+    build_name_copy = blame(request, "apicast-example-policy-copy")
+
+    github_params = {"AMP_RELEASE": amp_release,
+                     "BUILD_NAME": build_name_github,
+                     "IMAGE_STREAM_NAME": image_stream_name,
+                     "IMAGE_STREAM_TAG": "github"}
 
-    template_params = {"AMP_RELEASE": amp_release,
-                       "BUILD_NAME_EXAMPLE_POLICY": build_name_example_policy,
-                       "BUILD_NAME_CUSTOM_POLICY": build_name_custom_policies,
-                       "IMAGE_STREAM_APICAST_POLICY": image_stream_apicast_policy,
-                       "IMAGE_STREAM_AMP_APICAST_CUSTOM_POLICY": image_stream_amp_apicast_custom_policy
-                       }
+    copy_params = {"AMP_RELEASE": amp_release,
+                   "BUILD_NAME": build_name_copy,
+                   "TARGET_IMAGE_STREAM": image_stream_name,
+                   "TARGET_TAG": "latest",
+                   "EXAMPLE_POLICY_IMAGE_STREAM": image_stream_name,
+                   "EXAMPLE_POLICY_TAG": "github"}
 
     def _delete_builds():
-        openshift_client.delete_template(image_template, template_params)
+        openshift_client.delete_template(github_template, github_params)
+        openshift_client.delete_template(copy_template, copy_params)
 
     request.addfinalizer(_delete_builds)
```

Comment:
I'd just change this to make use of yield instead of finalizer

Reply:
I think the finalizer is better in this case as all of the commands below can sometimes fail which would leave many objects undeleted.

Reply:
right, `addfinalizer` seems appropriate here